### ISSUE TITLE
vtk output partially broken

### ISSUE THREAD

**Issue by sdrave** [pymor/pymor#284]
Executing the following code

``` python
from pymor.basic import *
p = Burgers2DProblem()
d, _ = discretize_nonlinear_instationary_advection_fv(p, nt=200)
U = d.solve(1.5)
d.visualize(U, filename='test')
```

creates vtk output which cannot be read by paraview 5.0.0. Opening one of the vtu files crashes paraview. Opening the pvd file leads to 

```
ERROR: In /build/paraview-dOO9Hm/paraview-5.0.0+dfsg1/ParaViewCore/VTKExtensions/Default/vtkXMLCollectionReader.cxx, line 416
vtkPVDReader (0x3df1a00): Could not determine the data type for the first dataset. Please make sure this file format is supported.
```

**Comment by sdrave** [comment]
Also, the error message in `gui.qt` regarding missing vtk still complains that `pyvtk` should be installed instead of the new library.

**Comment by sdrave** [comment]
pyvtk supposedly now also supports Python 3. Given the remaining issues, would it be an option to switch back to pytk, @renemilk?

**Comment by renemilk** [comment]
Does pyvtk still only write legacy format files? In that case I'd rather see the issue fixed. Btw, I've reached out to the pyevtk PyPi maintainer, who in turn contacted the original author. So we're getting an updated package there soonish.

**Comment by sdrave** [comment]
Probably it doesn't support `.vtu`. When did you contact the author? I cannot find an issue on https://bitbucket.org/pauloh/pyevtk.

**Comment by renemilk** [comment]
It wasn't I who contacted the author, I contacted the pypi maintainer.
https://bitbucket.org/somada141/pyevtk/issues/1/new-pypi-release#comment-31210186
https://bitbucket.org/pauloh/pyevtk/issues/30/error-using-imagetovtk#comment-None

**Comment by sdrave** [comment]
@renemilk, I have installed the latest version from https://bitbucket.org/pauloh/pyevtk/src, but the above code is still not working. No idea if this is a bug in pyevtk or pyMOR ..

**Comment by renemilk** [comment]
Pyvtk seems mostly python3 compatible, there's atm one open issue that we prolly won't run into. I've tried to transition to it, but it's not super-straigtforward and working directly. Pyvtk does indeed only write legacy vtk files. Which means we would get no time series ```.pvd``` to open with paraview, since those do not support having legacy files in them. 
I've also looked into what's wrong with the current pyevtk output loading in paraview, but as with anything VTK the output message is super unhelpful and there's very little useful info online.

**Comment by renemilk** [comment]
I've just checked this again and now I'm sure that this is a bug in pymor itself:
```
from pymor.basic import *
p = burgers_problem_2d(torus=False)
d, _ = discretize_instationary_fv(p, nt=2)
U = d.solve(1.5)
d.visualize(U, filename='test')
```
outputs files readable with paraview (5.1), while ```torus=True``` still results in segfaulting paraview. The .pvd is still unusable though.

So best guess would prolly be ```flatten_grid```? What do you think @sdrave ?

**Comment by sdrave** [comment]
@renemilk, I could fix the segfault in ef5eed9. However, a remaining issue is that the generated `.pvd` files do not work:

> ERROR: In /build/paraview-NOjbRJ/paraview-5.1.2+dfsg1/ParaViewCore/VTKExtensions/Default/vtkXMLCollectionReader.cxx, line 416
vtkPVDReader (0x557236426220): Could not determine the data type for the first dataset. Please make sure this file format is supported.

Could you take another look?

**Comment by sdrave** [comment]
Ok, so the only issue was, that the filenames in the `.pvd` needed to have the `.vtu` extension (6c7267d).
